### PR TITLE
feat(vow): VowShape, isVow

### PR DESCRIPTION
**Pull request by erights**
closes: #XXXX
refs: #9097 #9153

## Description

Initially extracted from #9097. 

Introduces an explicit `VowShape` --- a pattern for recognizing vows --- and an `isVow` predicate that uses that shape. Tightens `getPayload` so it only applies to vows that satisfy these conditions.

Also exports a new `VowTools` type.

### Security Considerations

none
### Scaling Considerations

none
### Documentation Considerations

none
### Testing Considerations

none
### Upgrade Considerations

The pattern is currently too specific to what vows look like in the current implementation of vow. The vow representation was explicitly designed to enable other vow representations in the future. It is unclear if the `VowShape` pattern should be loosened to accommodate that, at the risk of accepting non-vows, or if the `VowShape` pattern should evolve as vow representations evolve.

The current `VowShape` in this PR does already use `M.splitRecord` which should be adequate to anticipate the most likely evolutions of vow representations.

### CODE REVIEW

File: packages/vow/src/types.js [hunk 1]
```diff
@@ -122,3 +122,11 @@ export {};
  * @template [T=any]
  * @typedef {import('./types.js').ERef<T | import('./types.js').Vow<T>>} Specimen
  */
+
+/**
+ * @typedef {object} VowTools
+ * // TODO type according to prepareWatch returns
```
Comment (michaelfig) [inline packages/vow/src/types.js]: Do you mean something like:
```suggestion
 * @property {import('./watch.js).Watch} watch
 * @property {import('./when.js').When} when
 * @property {<T = any>() => VowKit<T>} makeVowKit
```

**Review by michaelfig — APPROVED** [review]
LGTM, after the types are fixed up.

File: packages/vow/src/types.js (hunk 1 above)
Comment (erights) [reply packages/vow/src/types.js]: I did indeed, thanks!

Done.

**Comment by erights** [comment]
Since this is approved before #9153, and they are otherwise independent, I'm going to reverse the staging order so this can go in first.

**Pull request merged**